lib/reek/context/code_context.rb

```diff
@@ -60,12 +62,13 @@ class CodeContext
       #     (send nil :puts
       #       (lvar :x)))
       def initialize(context, exp)
-        @context            = context
         @exp                = exp
         @children           = []
-        @visibility_tracker = VisibilityTracker.new
         @statement_counter  = StatementCounter.new
         @refs               = AST::ObjectRefs.new
+        @visibility         = :public
+
+        @context = context.append_child_context(self) if context
```

Comment:
Can we put a helpful comment in here when context can be nil? Those things always trip me up in our codebase. I then always pull out the debugger to have a lightbulb moment but I think we should rather document it ;)